compiler/src/openxla/compiler/nvgpu/Dialect/TritonFlow/Conversion/ConvertTritonToFlowDispatch.cpp

```diff
@@ -0,0 +1,351 @@
+#include "llvm/IR/Module.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/FormatVariadic.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/SourceMgr.h"
+#include "llvm/Support/raw_ostream.h"
+#include "mlir/Conversion/ArithToLLVM/ArithToLLVM.h"
+#include "mlir/Conversion/IndexToLLVM/IndexToLLVM.h"
+#include "mlir/Conversion/SCFToControlFlow/SCFToControlFlow.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/SymbolTable.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/Passes.h"
+#include "openxla/compiler/nvgpu/Dialect/TritonFlow/IR/TritonFlowOps.h"
+#include "triton/Conversion/TritonGPUToLLVM/TritonGPUToLLVMPass.h"
+#include "triton/Conversion/TritonToTritonGPU/TritonToTritonGPUPass.h"
+#include "triton/Dialect/Triton/IR/Dialect.h"
+#include "triton/Dialect/Triton/Transforms/Passes.h"
+#include "triton/Dialect/TritonGPU/Transforms/Passes.h"
+#include "triton/Target/LLVMIR/LLVMIRTranslation.h"
+#include "triton/Target/PTX/PTXTranslation.h"
+
+IREE_DEFINE_COMPILER_OPTION_FLAGS(
+    openxla::compiler::nvgpu::tritonflow::TritonOptions);
+
+namespace openxla::compiler::nvgpu::tritonflow {
+
+using namespace mlir;
+using namespace mlir::triton;
+using namespace mlir::iree_compiler;
+
+//===----------------------------------------------------------------------===//
+// TritonOptions
+//===----------------------------------------------------------------------===//
+
+void TritonOptions::bindOptions(OptionsBinder &binder) {
+  static llvm::cl::OptionCategory category("OpenXLA Triton compiler options");
+
+  binder.opt<int32_t>(
+      "openxla-triton-compute-capability", compute_capability,
+      llvm::cl::desc("Compute capability for compiling Triton programs."),
+      llvm::cl::cat(category));
+
+  binder.opt<int32_t>(
+      "openxla-triton-num-warps", num_warps,
+      llvm::cl::desc("Number of warps for compiling Triton programs."),
+      llvm::cl::cat(category));
+
+  binder.opt<int32_t>(
+      "openxla-triton-num-stages", num_stages,
+      llvm::cl::desc("Number of stages for compiling Triton programs."),
+      llvm::cl::cat(category));
+}
+
+//===----------------------------------------------------------------------===//
+// Compilation pipeline from Triton IR to LLVM IR
+//===----------------------------------------------------------------------===//
+
+void buildTritonCompilationPipeline(mlir::OpPassManager &pm,
+                                    const TritonOptions &opts) {
+  // Based on optimize_ttir() in
+  // @triton//:python/triton/compiler/compiler.py
```

Comment:
It'd be really good to have a better source of truth on this vs a bunch of freestanding passes that we maintain.